src/containers/reports/ducks/selectors.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  AdoptionReport,
+  StewardshipReport,
+  StewardshipReportTableEntry,
+} from './types';
+
+export const getCountAdoptedInPastWeek = (
+  adoptionReport: AdoptionReport | undefined,
+): number | undefined => {
+  return adoptionReport?.adoptionReport.filter(
+    (entry) =>
+      new Date(entry.dateAdopted) >
+      new Date(Date.now() - 7 * 24 * 60 * 60 * 1000),
+  ).length;
+};
+
+export const getStewardshipTableReport = (
+  stewardshipReport: StewardshipReport,
+): StewardshipReportTableEntry[] => {
+  return stewardshipReport.stewardshipReport.map((entry, idx) => {
+    const activitiesPerformed = [];
+
+    if (entry.watered) {
+      activitiesPerformed.push('watered');
+    }
+    if (entry.mulched) {
+      activitiesPerformed.push('mulched');
+    }
+    if (entry.cleaned) {
+      activitiesPerformed.push('cleaned');
+    }
+    if (entry.weeded) {
+      activitiesPerformed.push('weeded');
+    }
```

Comment:
⛏️ Seems like this should be pulled out into a helper. Its also a bit repetitive, we can instead for loop through the keys, check for their existence, and mutate. Or even better, filter a list of activities by those contained in `entry` like so:
```ts
const activitiesPerformed = ['watered', 'mulched', 'cleaned', 'weeded'].filter(activity => activity in entry)
```
Although using `in` to check for if a key exists in an object can be troublesome since it chases the prototype chain. You can also do:
```ts
const activitiesPerformed = ['watered', 'mulched', 'cleaned', 'weeded'].filter(activity => entry.hasOwnProperty(activity))
```

And and as a bonus we removed that nasty mutation 😊

Reply:
Was stumped for a while but figured out how to use filter and avoid the `TS7053` error (these checks are for whether the value of the key is true rather than if the key exists, so had to use `entry[activity]` instead of `entry.hasOwnProperty(activity)`. This gives you a type error unless you explicitly declare that the list is made up of valid keys).